File: frontend/src/types/audit-log.ts
import type { PaginationParams } from "@/types/pagination";

export type AuditActor = {
  email: string;
  first_name: string;
  id: string;
  last_name: string;
};

export type AuditLog = {
  action: string;
  created_at: string;
  entity_id: string;
  entity_type: string;
  id: string;
  user: AuditActor;
};

export type AuditLogsListParams = PaginationParams & {
  action?: string;
  entity_type?: string;
  search?: string;
  sort_direction?: "asc" | "desc";
};

File: backend/app/schemas/audit_log.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AuditActorResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    email: str
    first_name: str
    last_name: str


class AuditLogResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    action: str
    entity_type: str
    entity_id: UUID
    created_at: datetime
    user: AuditActorResponse
